packages/xforms-engine/src/client/ModelValueNode.ts

```diff
@@ -0,0 +1,40 @@
+import type { ValueNodeDefinition } from '../model/ValueNodeDefinition.ts';
+import type { BaseNode, BaseNodeState } from './BaseNode.ts';
+import type { GeneralParentNode } from './hierarchy.ts';
+import type { RootNode } from './RootNode.ts';
+import type { LeafNodeValidationState } from './validation.ts';
+
+export interface ModelValueNodeState extends BaseNodeState {
+	get label(): null;
+	get hint(): null;
+	get children(): null;
+	get valueOptions(): null;
+
+	/**
+	 * Reflects the current value of a {@link ModelValueNode}. This value may be
+	 * populated when a form is loaded, and it may be updated by certain
+	 * computations defined by the form.
+	 */
+	get value(): string;
+}
+
+export interface ModelValueDefinition extends ValueNodeDefinition {
+	readonly bodyElement: null;
+}
+
+/**
+ * A node which is:
+ *
+ * - model-only (i.e. it has no corresponding body element)
+ * - a leaf/value node (i.e. it has no element children; it may be defined in
+ *   the form's `<model>` as either an {@link Element} or {@link Attr})
+ */
+export interface ModelValueNode extends BaseNode {
+	readonly nodeType: 'model-value';
+	readonly appearances: null;
+	readonly definition: ModelValueDefinition;
+	readonly root: RootNode;
+	readonly parent: GeneralParentNode;
+	readonly currentState: ModelValueNodeState;
+	readonly validationState: LeafNodeValidationState;
```

Comment:
As with the last question, my instinct is to defer this for now. While we'll definitely have different behavior on your example form, I would want to see how it affects real user-defined forms before tailoring the behavior to contrived scenarios.

I think it's worth emphasizing that this will not be new behavior introduced by this PR. The same nodes will have the same `validationState` (as well as `currentState.readonly`, etc) with this change as they do on `main` today. The main things that are changing about them now are their `nodeType` and the absence of write methods.

I think I'll ultimately agree with you on the expected behavior here, but I think it's probably better treated as a separate scope of work. And I think it would be good to think about it in the context of the [questions raised above](https://github.com/getodk/web-forms/pull/174#discussion_r1693358593).